pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -1816,12 +1831,6 @@ func TestGetNodesToDaemonPods(t *testing.T) {
 			newPod("non-matching-owned-0-", "node-0", simpleDaemonSetLabel2, ds),
 			newPod("non-matching-orphan-1-", "node-1", simpleDaemonSetLabel2, nil),
 			newPod("matching-owned-by-other-0-", "node-0", simpleDaemonSetLabel, ds2),
-			func() *v1.Pod {
```

Comment:
Maybe move this up to `wantedPods` so we are clear to future readers that this behavior is required.

Reply:
SGTM